### PR TITLE
Docs Project Vision

### PR DESCRIPTION
**Pull request by SeanKilleen**
This PR aims to establish the overall approach of the docs project to get feedback from other project leads and core team members. The vision of the docs project will be referenced from the README.

### CODE REVIEW

**Comment by SeanKilleen** [comment]
@CharliePoole @rprouse @ChrisMaddock @jnm2 @mikkelbu @OsirisTerje feedback welcome! Particularly interested in:

* Any concerns you have with what's suggested
* Anything you think is missing in order to complete the vision statement (keep in mind that it's intended to be a vision statement of guiding principles, not a play by play of all process mechanics).

File: VISION.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Stakeholders / Collaborators
+
+The docs project supports and collaborates with the other NUnit projects and their respective leads -- for reference, those are currently:
+
+| Project | Lead |
+| ------- | ---- |
+| Framework | @rprouse and @jnm2 |
+| Console and Engine | @ChrisMaddock |
+| VS Adapters and Test Generator | @OsirisTerje|
+| NUnit Analyzers | @mikkelbu |
```
[the remark this answers is not in the document]
Comment (SeanKilleen) [reply VISION.md:16]: Good catch; will do. I ensured the people were listed but missed accounting for all projects. 👍

File: VISION.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* Typos, small content corrections, stylistic consistency, grammatical changes, and clear bug fixes are places where docs team members can make & merge changes without project member approval.
+* In general, we will ask everyone to aim to create an issue before submitting non-routine changes for consideration. This allows space for discussion ahead of changes.
+* We'll attempt to alert project teams or members to changes in their area in case they want to offer their insights. If no project team members respond within 48 hours, the docs team will work with the PR as they see fit (waiting on input, or merging). Should a project team have a concern after a merge, we'll remember our overall values and work with them to resolve any concerns as a high priority.
+* We'll give a wide berth and strong deference to project members / leads on their content changes. While we ask for the same PR process to make suggestions, we'll ensure that project leads and core team members have access to write to the docs in case they have a time-sensitive change to make; we trust them everyone operate in the spirit of the values of the project.
```
[the remark this answers is not in the document]
Comment (SeanKilleen) [reply VISION.md:37]: Yep, I'm aware. Goal is to get the vision in place and then work to make it so once others agree with the overall spirit and approach. 👍 Hoping to get this piece taken care of as one of the first actions.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ This repository serves the content that is found at <http://docs.nunit.org>.
 
 ![NUnit Documentation Build Process](https://github.com/nunit/docs/workflows/NUnit%20Documentation%20Build%20Process/badge.svg)
 
+## What is the Docs site? How does it work?
+
+The docs site is a project within the NUnit organization. [Read the vision at VISION.md](VISION.md) to understand more about how this document fits into the overall organization and how it supports the other projects.
```
[the remark this answers is not in the document]
Comment (SeanKilleen) [reply README.md:9]: Yes, thanks!

**Comment by SeanKilleen** [comment]
@ChrisMaddock 

> An idea to throw out there - thinking where we've come from, is making the docs accessible for contribution a key part of the vision?

Can you elaborate on this a little? I think I understand what you mean (and agree) but I want to be sure.

**Comment by ChrisMaddock** [comment]
What I was meaning, is that in my mind - the biggest benefit of your work here so far is that you've made the docs so much more accessible for everyone to contribute to. (The pretty new site is of course wonderful as well! 😄)

Making the docs accessible for contribution isn't mentioned in the vision statement. Of course, that's for you to decide if you think that's a key thing going forward or not - I just thought it might be worth a mention. 🙂

**Comment by SeanKilleen** [comment]
@ChrisMaddock I do think it's worth a mention. I was taking that for granted implicitly but I like the idea of adding it. Will do. Thanks!

File: VISION.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Who makes changes to the docs
+
+Given that there are many projects that feed into the docs and their members and leads work hard to maintain things, our goal is to be able to modify the docs without making it harder for project team members.
+
+* Any change has to pass the docs build process, linters, etc. before being merged. That process will continue to evolve to ensure the docs have a great build & release story.
+* Typos, small content corrections, stylistic consistency, grammatical changes, and clear bug fixes are places where docs team members can make & merge changes without project member approval.
+* In general, we will ask everyone to aim to create an issue before submitting non-routine changes for consideration. This allows space for discussion ahead of changes.
+* We'll attempt to alert project teams or members to changes in their area in case they want to offer their insights. If no project team members respond within 48 hours, the docs team will work with the PR as they see fit (waiting on input, or merging). Should a project team have a concern after a merge, we'll remember our overall values and work with them to resolve any concerns as a high priority.
```
Comment (CharliePoole) [inline VISION.md:37]: This wording seems to add a requirement for every project to respond to every PR within 48 hours, although that may not be what you intend. I think that's a great goal for any project. However...

1. It's not what happens now.
2. It may not be what each project wants to commit to.

Since the "Typos, small content corrections, etc." bullet point already took care of the easy stuff, this point is the one that deals with the hard stuff. It could easily take a few days just to understand whether a major rewrite of certain parts of the docs (e.g. parallel execution of tests) is even correct, without even considering how good the writing is.

Bottom line, I think that certain __kinds__ of changes as well as very broad changes will probably require some kind of approval of the project team. (After all, there are no tests to run to tell us if the docs are correct) 

That said, I'd be happy with just stating that there are some changes that must go through the project team and leaving it up to you to recognize those cases when they come up.

Comment (SeanKilleen) [reply VISION.md:37]: @CharliePoole that makes sense to me. You're correct that my intention was to say "if a project doesn't respond within 48 hours, it's left up to the discretion of the docs team" -- meaning, we might move forward, but are going to apply a lot of judgment on that.

I like the way you're approaching that phrasing and I believe I'll adopt that. 👍

Comment (SeanKilleen) [reply VISION.md]: ✅ Changes made to reflect this.

**Comment by SeanKilleen** [comment]
A though for consideration by the group --

I also want to make sure it's clear that one of the ways the docs project can help is by being pinged on any conversations / issues / PRs that might benefit from docs follow-up. This could be a direct mention in the issue itself, or could be a placeholder issue in the docs repo with a reference to the prior conversation.

Is that already clear and implicit to everyone here? Or is it worth making it clear as part of the overall vision?

**Comment by CharliePoole** [comment]
Here's an idea... for new features, the projects themselves have to come up with documentation. One thing you could do is monitor new features that got added - maybe using some automation - and try to correlate it with additions to the docs. If a big new feature comes along and you don't see anything happening in the docs, maybe fire off an issue __to that project__ reminding them that folks need documentation!

File: VISION.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+Its vision is to make it easier for NUnit's community to better understand and connect with NUnit's projects, and to make this documentation accessible to anyone who wishes to contribute back. The docs, in order to be successful, must be more than a hosting place for the separate projects' information. They must be a uniform and holistic journey for users to navigate the ecosystem and be confident that they understand things.
+
+## Stakeholders / Collaborators
+
+The docs project supports and collaborates with the other NUnit projects and their respective leads -- for reference, those are currently:
+
+| Project | Lead |
+| ------- | ---- |
+| Framework | @rprouse and @jnm2 |
```
Comment (jnm2) [inline VISION.md:13]: The framework lead is just @rprouse.

**Comment by jnm2** [comment]
> Is that already clear and implicit to everyone here? Or is it worth making it clear as part of the overall vision?

I think this is worth mentioning because that specific suggestion didn't form in my mind when I read through.

**Comment by SeanKilleen** [comment]
Added:

> * **We aim to be a support mechanism for project teams as much as possible.** We will suggest potential documentation improvements, but we also aim to enable project teams to ping us in conversations that might merit documentation to be included, or to make a suggestion that we can attempt to run with. We want to enable better docs, which means enabling other project teams to lean on us for support wherever possible.

**Comment by CharliePoole** [comment]
Here's what I would like to see as an overall process, while still remaining friendly to anyone who doesn't follow it exactly...

1. If something is undocumented, documented incorrectly or requires significant expansion, create an issue in the responsible project itself (i.e. framework, engine etc.).

2. Once the team has completed the work on (1) the PR should be created on the docs project with a cross-project "fixes" reference if that's possible.

3. If a user creates a docs issue on something covered by (1), transfer it immediately to the target project. @SeanKilleen can make that judgement, asking for advice if he needs it.

4. If a user creates a PR that should have gone the route of (1), require immediate review from the target project before anything else is done.

This reflects the notion that __in the extreme__ only the project team can really determine if the docs are correct and describe behavior they are wiling to support. Of course, in specific cases, Sean and anyone else added to the docs project may be able to make a judgement call. IMHO cooperation happens best when people are allowed to make judgements and don't get beaten up when the judgement turns out to have been mistaken.

What I'd like to avoid is adding documentation that says "X works this way..." and then have a bug filed six months later because X doesn't actually work that way in edge cases and possibly was never intended to work that way. We've had a lot of feature creep in the past due to stuff like that.

**Comment by mikkelbu** [comment]
I also like the vision and Charlie's proposed workflow. Regarding the workflow I think that most doc issues will actually appear as in item 3 as most users don't distinguish (or know the difference) between framework, engine, adapter ... So I guess that most issues will be created in this repo. Ps. cross repo auto closure is possible https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#closing-an-issue-in-a-different-repository.

**Comment by SeanKilleen** [comment]
Having not heard any overall concerns on the overall vision and several approvals, I'm going to go ahead and merge this so that it's out there.

With that said, follow-up issues and PRs are welcome at anytime; the vision, like the project, is a living thing. 👍
